src/app/components/navbar/navbar.component.ts

```diff
@@ -15,13 +20,56 @@ import {logout} from '../../user/actions/user.actions';
 export class NavbarComponent implements OnInit {
   signInUser$: Observable<User>;
   userFullName$: Observable<string>;
+  transactions$: Observable<Transaction[]>;
+  user: User;
+  alerts: Alert[] = [];
+  displayNotification: boolean = false;
 
-  constructor(private router: Router, private store: Store<UserState>) {
+  constructor(private router: Router,
+              private store: Store<UserState>,
+              private transactionService: TransactionsService,
+              private userService: UsersService) {
   }
 
   ngOnInit(): void {
-    this.signInUser$ = this.store.pipe(select(getUser));
+    this.signInUser$ = this.store.pipe(select(getUser)).pipe(tap(user => this.user = user));
     this.userFullName$ = this.signInUser$.pipe(map(user => user ? `${user.firstName} ${user.lastName}` : 'no name'));
+    this.createAlerts();
+  }
+
+  private createAlerts(): void {
+    this.transactions$ = this.transactionService.getNewTransactions$();
+    this.transactionService.getNewTransactions$().pipe(mergeMap((transactions: Transaction[]) =>
+      forkJoin(
+        transactions.map((transaction: Transaction) => forkJoin({
+          transactionId: of(transaction.id),
+          description: of(this.getDescription(transaction.status)),
+          trader: this.userService.getById$(transaction.operatedBy).pipe(first()),
+          date: of(!!transaction.transactionCompleteDate ? transaction.transactionCompleteDate.toDate() : transaction.offerDate.toDate())
+        })))
+    )).subscribe((alerts: Alert[]) => {
+      this.alerts = alerts;
```

Comment:
can make it one line without assignment 
this.alerts = alert.sort((a: Alert, b: Alert) => b.date.getTime() - a.date.getTime());

Reply:
done